packages/test_track/lib/src/networking/interceptors/retry_interceptor.dart

```diff
@@ -1,31 +1,42 @@
 import 'dart:async';
 
 import 'package:dio/dio.dart';
-import 'package:test_track/src/networking/http_client.dart';
+import 'package:sturdy_http/sturdy_http.dart';
+import 'package:test_track/src/networking/extras.dart';
 import 'package:test_track/src/networking/interceptors/retry_options.dart';
 
+/// Function that must return the [SturdyHttp] instance that will
+/// handle the retry.
+typedef SturdyHttpGetter = SturdyHttp Function();
+
 /// {@template retry_interceptor}
 /// An [Interceptor] responsible for performing retry
 /// operations in the case of a failed network request
 /// {@endtemplate}
 class RetryInterceptor extends Interceptor {
-  final Dio _dio;
+  final SturdyHttpGetter _clientGetter;
   final RetryOptions _retryOptions;
 
   /// {@macro retry_interceptor}
   RetryInterceptor({
-    required Dio dio,
+    required SturdyHttpGetter clientGetter,
     RetryOptions? retryOptions,
-  })  : _dio = dio,
+  })  : _clientGetter = clientGetter,
         _retryOptions = retryOptions ?? RetryOptions();
 
   @override
-  Future onError(DioError err, ErrorInterceptorHandler handler) async {
+  Future<void> onError(
+      DioException err, ErrorInterceptorHandler handler) async {
     var extra =
         RetryOptions.fromRequestOptions(err.requestOptions) ?? _retryOptions;
     final isIdempotent =
-        err.requestOptions.extra[HttpClient.isIdempotentOptionsKey] as bool? ??
-            false;
+        err.requestOptions.extra[isIdempotentOptionsKey] as bool? ?? false;
+    final originalMethod = err.requestOptions.method;
+    final capitalizedMethod = StringBuffer()
+      ..write(originalMethod.substring(0, 1).toUpperCase())
+      ..write(originalMethod.substring(1).toLowerCase());
```

Comment:
😲

Reply:
We actually assign it to a `NetworkRequestType` which is just our `enum` representation of the HTTP method! We then use that type as the `type` parameter to `RawRequest`. Without `RawRequest` (introduced [here](https://github.com/Betterment/sturdy_http/pull/3)), we'd have to `switch` on the `NetworkRequestType` and produce either a `GetRequest`, `PostRequest`, `PutRequest` or `DeleteRequest`, all of which would contain the exact same metadata but with a different HTTP method!

Reply:
oooooooh !!!! that's sick